### ISSUE TITLE
High number of duplicates with -so option

### ISSUE THREAD

**Issue by carlosmag** [amplab/snap#126]
Hi,
I am getting ≃ 2.5 more duplicates marked in `snap paired with -so parameter` than with `snap paired without -so option + pipe to samtools markdup or picard MarkDuplicates`.
Stats obtained with _samtools flagstat_: 646886 vs 264372 duplicates.

Is there any issue with _snap_ or interoperability with other tools?

Test genome and bam files [here](https://mega.nz/folder/dERWDI6I#37xE-kQd_NjtP06-xEyVAg)
Reference genome [here](https://figshare.com/articles/Genome_of_the_inferred_most_recent_common_ancestor_of_the_Mycobacterium_tuberculosis_complex/11500980/1)

`SNAP version 1.0dev.102`
`samtools 1.10`
`Picard 2.23.0`

**Comment by bolosky** [comment]
We’ve been working on getting concordance with Picard markdup right now.  Arun is doing the work, maybe he can comment.

**Comment by arun-sub** [comment]
We are aware of the differences in SNAP duplicate marking with respect to Picard MarkDuplicates. The differences in SNAP are mainly due to: (1) not taking soft-clips into account and (2) not marking singleton read duplicates (i.e., when only read in the pair is mapped).

I tried your dataset on a version of a new release that we are currently working on and have seen significantly fewer differences (~1000-2000 reads w/ Picard Markdup).  Unfortunately the new version of the code has diverged significantly from the current version that you are using, making it difficult for us to backpatch these changes onto your version, The new release will include bug fixes as well as support for affine gap scoring and performance improvements. We plan to release this in the next few months before the end of summer.

I will keep this issue open and will update here once we have resolved the discrepancies.

--Arun
